nxt/stage.py

```diff
@@ -4046,13 +4060,16 @@ def run(runtime_layer, stage=None, rt_node=None, custom_code=None):
     console.running_lines = lines
     console.globals['self'] = frame_node
     runtime_layer.running = True
+    exit_exception = None
     try:
         console.runcode(_code)
     except GraphError:
         if not console.run_as_global:
             clean_globals(lines, good_keys, console.globals)
         runtime_layer.running = False
         raise
+    except (ExitNode, ExitGraph) as exit_exception:
+        pass
```

Comment:
Did you experiment with an reworking this whole try?
Something like this I think would clean up some of the existing mess as well. What do you think?
```
try:
    console.runcode(_code)
except (GraphError, ExitNode, ExitGraph):
    raise
finally:
    runtime_layer.running = False
    if not console.run_as_global:
    ...
```

Reply:
Yeah I thought of this, I'm not convinced we want to do the cache value comparison (LN 4077) in the event of a `GraphError`. I can't remember if we had a good reason for _not_ doing that.